client/src/components/Header.tsx

```diff
@@ -0,0 +1,115 @@
+import AppBar from "@mui/material/AppBar";
+import Link from "@mui/material/Link";
+import InputBase from "@mui/material/InputBase";
+import SearchIcon from "@mui/icons-material/Search";
+import { styled, alpha } from "@mui/material/styles";
+import Toolbar from "@mui/material/Toolbar";
+import MenuItem from "@mui/material/MenuItem/MenuItem";
+import Menu from "@mui/material/Menu";
+import React from "react";
+import AccountCircle from "@mui/icons-material/AccountCircle";
+import IconButton from "@mui/material/IconButton/IconButton";
+
+const Search = styled("div")(({ theme }) => ({
+  position: "relative",
+  borderRadius: theme.shape.borderRadius,
+  backgroundColor: alpha(theme.palette.common.white, 0.15),
+  "&:hover": {
+    backgroundColor: alpha(theme.palette.common.white, 0.25),
+  },
+  marginLeft: "auto",
+  marginRight: "auto",
+  width: "auto%",
+}));
+
+const SearchIconWrapper = styled("div")(({ theme }) => ({
+  padding: theme.spacing(0, 2),
```

Comment:
Should this be a theme as well? Here and elsewhere.

Reply:
This is how the mui docs said to do it: https://mui.com/customization/spacing/ so everything stays consistent if we decide to customize our theme.

They also did a similar approach in some of their other examples.

Reply:
What I'm saying is should we. I've never seen theming used down to this level since it's very, very low level. It's highly unlikely that there would be a global/theme level spacing change unless its for something like spacing between cards for example. But if you rely on using the theme everywhere, it'll break stuff.